apiserver/plane/app/views/project/base.py

```diff
@@ -212,20 +212,20 @@ def list(self, request, slug):
     )
     def retrieve(self, request, slug, pk):
         project = (
-            self.get_queryset()
-            .filter(
-                project_projectmember__member=self.request.user,
-                project_projectmember__is_active=True,
-            )
-            .filter(archived_at__isnull=True)
-            .filter(pk=pk)
-        ).first()
+            self.get_queryset().filter(archived_at__isnull=True).filter(pk=pk).first()
+        )
 
         if project is None:
             return Response(
                 {"error": "Project does not exist"}, status=status.HTTP_404_NOT_FOUND
             )
 
+        if project.member_role is None:
```

Comment:
@sangeethailango This is not the right check to do here. We need to verify if the current use is a project member of this project or not